File: app/src/main/java/io/github/anonymous/pepper_realtime/di/NetworkModule.java
package io.github.anonymous.pepper_realtime.di;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.components.SingletonComponent;
import io.github.anonymous.pepper_realtime.network.RealtimeSessionManager;

@Module
@InstallIn(SingletonComponent.class)
public class NetworkModule {

    @Provides
    @Singleton
    public RealtimeSessionManager provideRealtimeSessionManager() {
        return new RealtimeSessionManager();
    }
}

File: app/src/main/java/io/github/anonymous/pepper_realtime/di/NetworkModule.kt
package io.github.anonymous.pepper_realtime.di

import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import io.github.anonymous.pepper_realtime.network.RealtimeSessionManager
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
object NetworkModule {

    @Provides
    @Singleton
    fun provideRealtimeSessionManager(): RealtimeSessionManager {
        return RealtimeSessionManager()
    }
}

